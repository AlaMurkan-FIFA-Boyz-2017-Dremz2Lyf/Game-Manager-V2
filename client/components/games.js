import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  ListGroup
} from 'react-bootstrap';

import Game from './game';

import { fetch } from '../actions';

export class Games extends Component {

  componentDidMount() {
    this.props.fetch('games', {type: 'tournament', id: this.props.tournament});
  }

  renderGames() {
    let { games = {} } = this.props;
    return Object.keys(games).map(id => (
      <Game key={id} game={games[id]} />
    ));
  }

  render() {
    return (
      <ListGroup>
        {this.renderGames()}
      </ListGroup>
    );
  }
}

export const mapStateToProps = ({ data }) => ({
  games: data.games
});

export default connect(mapStateToProps, { fetch })(Games);

Games.propTypes = {
  tournament: React.PropTypes.string
};
