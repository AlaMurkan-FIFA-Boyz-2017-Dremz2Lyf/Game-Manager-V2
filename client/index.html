<html>
<head>
  <title>Game Manager</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans|Oswald|PT+Sans">
</head>
<body>
  <div id="app"></div>
</body>
</html>
