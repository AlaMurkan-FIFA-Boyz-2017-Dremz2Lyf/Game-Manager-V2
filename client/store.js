
// Make our store exportable for using in tests/and production

import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index.js';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(rootReducer, undefined, composeEnhancers(
  applyMiddleware(thunk)
));
