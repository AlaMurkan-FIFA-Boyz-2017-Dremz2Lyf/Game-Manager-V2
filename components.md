 Component | State | Actions
---|---|---
 Home | None? | None?
 AllTimeStats | {currentList: ongoing/finished/new} | None?
