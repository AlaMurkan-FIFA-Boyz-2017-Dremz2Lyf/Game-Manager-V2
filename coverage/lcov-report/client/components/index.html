<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> client/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>95/95</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>25/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>51/51</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>91/91</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="create_player.js"><a href="create_player.js.html">create_player.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="create_tournament.js"><a href="create_tournament.js.html">create_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	</tr>

<tr>
	<td class="file high" data-value="game.js"><a href="game.js.html">game.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="games.js"><a href="games.js.html">games.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="home.js"><a href="home.js.html">home.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="play_tournament.js"><a href="play_tournament.js.html">play_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="player_form.js"><a href="player_form.js.html">player_form.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list.js"><a href="player_list.js.html">player_list.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list_item.js"><a href="player_list_item.js.html">player_list_item.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="tournament.js"><a href="tournament.js.html">tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="tournaments.js"><a href="tournaments.js.html">tournaments.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Mar 10 2017 17:46:58 GMT-0600 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
