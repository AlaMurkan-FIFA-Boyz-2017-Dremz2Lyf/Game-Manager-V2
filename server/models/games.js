const model = require('./lib/knex_model');

let games = module.exports = model.create('games', {
  tableName: 'games'
});
