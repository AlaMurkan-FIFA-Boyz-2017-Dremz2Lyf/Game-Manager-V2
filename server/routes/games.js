const express = require('express');
const router = express.Router();
const _ = require('../utilities.js');
const games = require('../models/games');

// NOTE: All this ' is a placeholder for a database.

let gameTable = [
  {id: 1, p1: 1, p2: 2, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null, p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournamentId: 1, status: 'created', createdAt: '2016-12-25T19:31:48'},
  {id: 2, p1: 1, p2: 3, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null, p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournamentId: 1, status: 'created', createdAt: '2016-12-25T19:31:48'},
  {id: 3, p1: 2, p2: 3, p1Score: null, p2Score: null, p1Shots: null, p2Shots: null, p1Poss: null, p2Poss: null, p1OnGoal: null, p2OnGoal: null, tournamentId: 1, status: 'created', createdAt: '2016-12-25T19:31:48'}
];

/*********************************************/

router.name = 'games';

router.get('/', (req, res) => {
  games.all().then(rows => {
    console.log('games rows:', rows);
  });
  // Sending fake data until the database is up.
  if (req.query.type === 'game') {
    res.send([gameTable[req.query.id - 1]]);
  } else if (req.query.type === 'tournament') {
    let response = gameTable.filter(game => game.tournamentId === req.query.id);
    res.send(response);
  } else {
    res.send(gameTable);
  }

});

router.post('/', (req, res) => {

  let players = req.body.players;
  let tournamentId = req.body.tournamentId;

  let response = {
    gamesCreated: 0
  };

  // placeholders for the 'interaction with the Database'
  // ********************************
  if (players.length < 3) {
    let game = _.createGame(players[0], players[1]);

    game.id = gameTable.length;

    response.gamesCreated = 1;

    gameTable.push(game);
  } else {
    let newGames = _.createGames(players, tournamentId);

    gameTable = gameTable.concat(newGames);

    gameTable.forEach((game, index) => {
      game.id = game.id || index + 1;
    });

    response.gamesCreated = gameTable.length;
  }
// **********************************
  res.status(201).send(response);
});

router.put('/', (req, res) => {

  let index = req.body.id - 1;

  gameTable[index] = req.body;

  let response = {id: req.body.id};
  res.status(202).send(response);
});

module.exports = router;
