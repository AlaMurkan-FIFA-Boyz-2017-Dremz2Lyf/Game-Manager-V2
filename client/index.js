// Dependancies
import React from 'react';
import ReactDOM from 'react-dom';
import { ProviderÂ } from 'react-redux';

// css
import '../styles/styles.css';

// Bring in our router
import Routes from './router';

// Grab our root reducer and our store from their files
import rootReducer from './reducers/index.js';
import { store } from './store';

// Found this... Needed for hot module reloading.
if (module.hot) {
  module.hot.accept();
}

ReactDOM.render(
  <Provider store={store}>
    <Routes />
  </Provider>,
  document.getElementById('app')
);
