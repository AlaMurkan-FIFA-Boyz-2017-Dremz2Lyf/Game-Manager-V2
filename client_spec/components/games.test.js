import React from 'react';
import { Provider } from 'react-redux';

import { store } from '../../client/store';

import { Games } from '../../client/components/games';

describe('Games Component', () => {
  let mockFetch = jest.fn();
  let mockUpdate = jest.fn();
  let wrapper = shallow(
    <Games
      tournament={'1'}
      games={mockData.games.slice(0, 3)}
      update={mockUpdate}
    />
  );

  let mounted = mount(
    <Provider store={store}>
      <Games
        tournament={'1'}
        fetch={mockFetch}
      />
    </Provider>
  );

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not fail with no props', () => {
    let empty = shallow(<Games />);

    expect(empty).toMatchSnapshot();
  });

  it('should render a list of Game components', () => {

    expect(wrapper.find('Connect(Game)').length).toBe(3);
  });

  it('should call fetch when it\'s mounted', () => {

    expect(mockFetch).toHaveBeenCalled();
    expect(mockFetch.mock.calls[0]).toEqual(['games', {type: 'tournament', id: '1'}]);
  });

});
