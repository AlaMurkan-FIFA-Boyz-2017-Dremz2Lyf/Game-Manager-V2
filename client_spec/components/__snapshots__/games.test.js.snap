exports[`Games Component should match the snapshot 1`] = `
<ListGroup
  bsClass="list-group">
  <Connect(Game)
    game={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 1,
        "p1": 1,
        "p1OnGoal": null,
        "p1Poss": null,
        "p1Reds": 0,
        "p1Score": null,
        "p1Shots": null,
        "p1Yellows": 0,
        "p2": 2,
        "p2OnGoal": null,
        "p2Poss": null,
        "p2Reds": 0,
        "p2Score": null,
        "p2Shots": null,
        "p2Yellows": 0,
        "status": "created",
        "tournament": 1,
        "updatedAt": null,
      }
    } />
  <Connect(Game)
    game={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 2,
        "p1": 1,
        "p1OnGoal": null,
        "p1Poss": null,
        "p1Reds": 0,
        "p1Score": null,
        "p1Shots": null,
        "p1Yellows": 0,
        "p2": 3,
        "p2OnGoal": null,
        "p2Poss": null,
        "p2Reds": 0,
        "p2Score": null,
        "p2Shots": null,
        "p2Yellows": 0,
        "status": "created",
        "tournament": 1,
        "updatedAt": null,
      }
    } />
  <Connect(Game)
    game={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 3,
        "p1": 2,
        "p1OnGoal": null,
        "p1Poss": null,
        "p1Reds": 0,
        "p1Score": null,
        "p1Shots": null,
        "p1Yellows": 0,
        "p2": 3,
        "p2OnGoal": null,
        "p2Poss": null,
        "p2Reds": 0,
        "p2Score": null,
        "p2Shots": null,
        "p2Yellows": 0,
        "status": "created",
        "tournament": 1,
        "updatedAt": null,
      }
    } />
</ListGroup>
`;

exports[`Games Component should not fail with no props 1`] = `
<ListGroup
  bsClass="list-group" />
`;
